overcooked_ai_py/agents/view_agent.py

```diff
@@ -0,0 +1,357 @@
+                  thinking_prob=thinking_prob0, prob_pausing=prob_pausing0,
+                  path_teamwork=path_teamwork0, rationality_coefficient=rat_coeff0)
+    # a0 = ToMModel(mlp, player_index=0, perseverance=0.9, teamwork=1, retain_goals=0.9,
+    #                                  wrong_decisions=0.02, thinking_prob=1, path_teamwork=1, rationality_coefficient=2)
+    # a1 = GreedyHumanModel_pk(mlp, player_index=0, perseverance=0.8)
+    # a1 = RandomAgent(mlp)
+    print(perseverance0, teamwork0, retain_goals0, wrong_decisions0, thinking_prob0, prob_pausing0, path_teamwork0,
+          rat_coeff0)
+    # print('Player 0: teamwork: {:.1f}, retain: {:.1f}, wrong dec: {:.1f}'.format(teamwork0, retain_goals0, wrong_decisions0))
+    return AgentPair(a0, a1)
+
+
+if __name__ == "__main__" :
+    """
+    
+    """
+    parser = ArgumentParser()
+    # parser.add_argument("-l", "--fixed_mdp", dest="layout",
+    #                     help="name of the layout to be played as found in data/layouts",
+    #                     required=True)
+    parser.add_argument("-l", "--layout",
+                        help="layout, (Choose from: sim, sc1, sch, uni, ran1, ran0)", required=True)
+    print("\n****************************************\nNOTE: To watch play in debug, put breakpoint in "
+          "overcooked_env.OvercookedEnv.run_agents\n*****************************************\n")
+    args = parser.parse_args()
+    layout = args.layout
+
+    np.random.seed(41)
+
+    # To print the agent's decisions:
+    logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.ERROR)
+
+    n, s = Direction.NORTH, Direction.SOUTH
+    e, w = Direction.EAST, Direction.WEST
+    stay, interact = Action.STAY, Action.INTERACT
+    P, Obj = PlayerState, ObjectState
+
+    DISPLAY = True
+    start_order_list = ["any"]*20
+    horizon = 2000
+    explosion_time = 500
+    r_shaping = 0
+
+    no_counters_params = {
+        'start_orientations': False,
+        'wait_allowed': False,
+        'counter_goals': [],
+        'counter_drop': [],
+        'counter_pickup': [],
+        'same_motion_goals': True
+    }
+
+    if layout == 'uni':
+        start_state = OvercookedState([P((2, 2), n), P((5, 2), n)], {}, order_list=start_order_list)
+    elif layout == 'sc1':
+        start_state = OvercookedState([P((1, 2), n), P((1, 1), n)], {}, order_list=start_order_list)
+    elif layout == 'ran1':
+        start_state = OvercookedState([P((1, 2), n), P((1, 1), n)], {}, order_list=start_order_list)
+    elif layout == 'ran0':
+        start_state = OvercookedState([P((1, 2), n), P((3, 2), n)], {}, order_list=start_order_list)
+    elif layout == 'ran3':
+        start_state = OvercookedState([P((1, 2), n), P((6, 2), n)], {}, order_list=start_order_list)
+    else:
+        start_state = OvercookedState([P((2, 2), n), P((2, 1), n)], {}, order_list=start_order_list)
+
+    if layout == 'sc1':
```

Comment:
If you want to commit this to overcooked_ai, it would be nice to remove all the repeats in the code first (have a way to abstract this rather than having 5 if statements with a lot of repeated code). Alternatively, you should probably use your project repo: I'll be less rigid in enforcing nice coding practice there, as it will mostly just be affecting you. This file is a great idea though! 😄

Reply:
hehe oh dear, this code of mine is exceptionally bad!! I've moved it to my own repo now (and also sorted it out!)